src/code_197F40.c

```diff
@@ -472,262 +516,1549 @@ ApiStatus GetActorPos(ScriptInstance* script, s32 isInitialCall) {
+        actorPart->partOffset.x = x;
+        actorPart->partOffset.y = y;
+        actorPart->partOffset.z = z;
+    } else {
+        actorPart->visualOffset.x = x;
+        actorPart->visualOffset.y = y;
+        actorPart->visualOffset.z = z;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus AddPartDispOffset(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorID actorID = get_variable(script, *args++);
+    ActorPart* actorPart;
+    s32 partIndex;
+    f32 x, y, z;
+
+    if (actorID == ActorID_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    partIndex = get_variable(script, *args++);
+
+    x = get_float_variable(script, *args++);
+    y = get_float_variable(script, *args++);
+    z = get_float_variable(script, *args++);
+    actorPart = get_actor_part(get_actor(actorID), partIndex);
+
+    if (!(actorPart->flags & 0x100000)) {
+        actorPart->partOffset.x += x;
+        actorPart->partOffset.y += y;
+        actorPart->partOffset.z += z;
+    } else {
+        actorPart->visualOffset.x += x;
+        actorPart->visualOffset.y += y;
+        actorPart->visualOffset.z += z;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_8026BF48(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = BATTLE_STATUS;
+    BattleStatus* battleStatus2 = BATTLE_STATUS;
+    Bytecode* args = script->ptrReadPos;
+
+    if (get_variable(script, *args++) != 0) {
+        battleStatus->unk_8C++;
+    } else if (battleStatus->unk_8C > 0) {
+        battleStatus2->unk_8C--;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus GetActorVar(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorID actorID = get_variable(script, *args++);
+    s32 var1;
+    s32 a2;
+
+    if (actorID == ActorID_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    var1 = get_variable(script, *args++);
+    a2 = *args++;
+
+    set_variable(script, a2, get_actor(actorID)->varTable[var1]);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus SetActorVar(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorID actorID = get_variable(script, *args++);
+    s32 index;
+    s32 val;
+
+    if (actorID == ActorID_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    index = get_variable(script, *args++);
+    val = get_variable(script, *args++);
+
+    get_actor(actorID)->varTable[index] = val;
+
+    return ApiStatus_DONE2;
+}
+
+#ifdef NON_MATCHING
+ApiStatus AddActorVar(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorID actorID = get_variable(script, *args++);
+    s32 index;
+    s32 val;
+
+    if (actorID == ActorID_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    index = get_variable(script, *args++);
+    val = get_variable(script, *args++);
+
+    get_actor(actorID)->varTable[index] += val;
```

Comment:
for this nonmatching, have you tried a temp for the actor (this mattered for me in one case already) and also what about
``varTable[get_variable(script, *args++)] += get_variable(script, *args++);``